core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -60,6 +131,31 @@ protected NamedWriteableRegistry provideNamedWritableRegistry() {
     }
 
     /**
+     * Test that a malformed JSON completion suggestion request fails.
+     */
+    public void testMalformedJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
+        final XContentBuilder suggest = jsonBuilder().startObject()
+                                            .startObject("bad-payload")
+                                            .field("prefix", "sug")
+                                            .startObject("completion")
+                                            .field("field", field)
+                                            .startArray("payload")
+                                            .startObject()
+                                            .field("payload", "field")
+                                            .endObject()
+                                            .endArray()
+                                            .endObject()
+                                            .endObject().endObject();
+        try {
+            final SuggestBuilder suggestBuilder = SuggestBuilder.fromXContent(newParseContext(suggest), suggesters);
+            fail("Should not have been able to create SuggestBuilder from malformed JSON: " + suggestBuilder);
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), equalTo("Suggestion must have name"));
```

Comment:
I this the same failure reason like in the original IT test that you moved here? This looks like its a parsing error that is new and comes from SuggestBuilder, but I suspect that the original test was checking something slightly different. I might be mistaken though, will look at what the original test does on master again.